### PR TITLE
drt: pin access split

### PR DESCRIPTION
**Pull request by bnmfw**
This PR creates three new PA files, one for each of the `FlexPA_prep.cpp` responsibilities:
- Generating Access Points for Unique Instances: `FlexPA_acc_point.cpp`
- Generating Access Patterns for Unique Instances: `FlexPA_acc_patterns.cpp`
- Generating Access Patterns to Rows of Instances: `FlexPA_row_patterns.cpp`

Some functions that couldn't be easily separated were moved to `FlexPA.cpp`, this might be due to being dependant to more than one of the new files or more than one of the new files depending on it.

I copied imports and copyright to new files. I commented any import that did not stop the project from compiling and running.

`FlexPA_acc_point.cpp` is still big 1517 LOC so it might have to be split even further.

I also deleted `enclosesOnTrackPlanarAccess` as it was never used. Here is the full code in case this ever is relevant:
```
bool FlexPA::enclosesOnTrackPlanarAccess(
    const gtl::rectangle_data<frCoord>& rect,
    frLayerNum layer_num)
{
  frCoord low, high;
  frLayer* layer = getDesign()->getTech()->getLayer(layer_num);
  if (layer->isHorizontal()) {
    low = gtl::yl(rect);
    high = gtl::yh(rect);
  } else if (layer->isVertical()) {
    low = gtl::xl(rect);
    high = gtl::xh(rect);
  } else {
    logger_->error(
        DRT,
        1003,
        "enclosesPlanarAccess: layer is neither vertical or horizontal");
  }
  const auto& tracks = track_coords_[layer_num];
  const auto low_track = tracks.lower_bound(low);
  if (low_track == tracks.end()) {
    logger_->error(DRT, 1004, "enclosesPlanarAccess: low track not found");
  }
  if (low_track->first > high) {
    return false;
  }
  auto high_track = tracks.lower_bound(high);
  if (high_track != tracks.end()) {
    if (high_track->first > high) {
      high_track--;
    }
  } else {
    logger_->error(DRT, 1005, "enclosesPlanarAccess: high track not found");
  }
  if (high_track->first - low_track->first > (int) layer->getPitch()) {
    return true;
  }
  if (low_track->first - (int) layer->getWidth() / 2 < low) {
    return false;
  }
  if (high_track->first + (int) layer->getWidth() / 2 > high) {
    return false;
  }
  return true;
}
```

### CODE REVIEW

**Comment by maliberty** [comment]
@bnmfw what's the status of this pr?

**Comment by bnmfw** [comment]
> @bnmfw what's the status of this pr?

I'm waiting for #5872 to be approved to continue.

File: src/drt/src/pa/FlexPA_acc_pattern.cpp [hunk 1]
```diff
@@ -0,0 +1,895 @@
...
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE REGENTS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
```
Comment (osamahammad21) [inline src/drt/src/pa/FlexPA_acc_pattern.cpp:27]: @bnmfw update the copyrights to the Precision Innovations copyrights. check src/drt/src/db/infra/KDTree.hpp

File: src/drt/src/pa/FlexPA_row_pattern.cpp [hunk 2]
```diff
@@ -0,0 +1,456 @@
...
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE REGENTS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
```
Comment (osamahammad21) [inline src/drt/src/pa/FlexPA_row_pattern.cpp:27]: ditto

**Review by osamahammad21 — APPROVED** [review]
I didn't look much into the code assuming that it's just moving around into new classes.

File: src/drt/src/pa/FlexPA_acc_pattern.cpp (hunk 1 above)
Comment (maliberty) [reply src/drt/src/pa/FlexPA_acc_pattern.cpp:27]: Unless it is new code we can add to the copyright but should not remove the old one.

**Pull request merged**